controllers/health/doguChecker.go

```diff
@@ -0,0 +1,162 @@
+package health
+
+import (
+	"context"
+	"fmt"
+	"github.com/cloudogu/cesapp-lib/core"
+	"github.com/cloudogu/cesapp-lib/registry"
+	k8sv1 "github.com/cloudogu/k8s-dogu-operator/api/v1"
+	"github.com/hashicorp/go-multierror"
+	v1 "k8s.io/api/apps/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// NewDoguHealthError creates a new dogu health error.
+func NewDoguHealthError(err error) *DoguHealthError {
+	return &DoguHealthError{err: err}
+}
+
+// DoguHealthError is a dogu validation error. Instances can be unwrapped. Instances can be type asserted.
+type DoguHealthError struct {
```

Comment:
I thought the `DoguHealthError` should implement the `RequeueableError` interface?